pywin32_testall.py

```diff
@@ -20,7 +20,10 @@ def run_test(script, cmdline_rest=""):
     popen = subprocess.Popen(cmd, shell=True, cwd=dirname,
                              stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
     data = popen.communicate()[0]
-    sys.stdout.buffer.write(data)
+    if "buffer" in dir(sys.stdout):
```

Comment:
wouldn't `hasattr()` be better here? And can you please remind me why we use .buffer in the first place? :)

Reply:
The gurus of Python suggest that hasattr() is usually implemented with a try/except, so it is more efficient to put the try/except in your original code. (Easier to Ask Forgiveness than Permission.)
  Having said that, I have thoroughly messed up this PR with my PR #1445 where I removed the `.buffer` and put in Python3 support.  
  If we really do need `.buffer` I suggest:
```
    if sys.version_info > (3,):
        data =  data.decode('latin-1')
    try:
        sys.stdout.buffer.write(data)
    except AttributeError:
        sys.stdout.write(data)
        sys.stdout.flush()
```

Reply:
> The gurus of Python suggest that hasattr() is usually implemented with a try/except, so it is more efficient to put the try/except in your original code.

I'm not too bothered with efficiency here - in this scenario I think readability trumps it. It's also unclear as written - do you expect 'stdout', 'buffer', or 'write' to be the missing attribute? Or possibly all 3? (That's a somewhat rhetorical question obviously :)

@thopiekar, did you plan on revising this?